### DISCUSSION TITLE
Next full estático, sem consultar as apis no client-side novamente, tem como?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1645] [Perguntas e Respostas]
Fala pessoal, 
feliz 2020 a todos, de muitas conquistas e felicidades!

Terminei um curso e o get started do next.js, e de hoje até o final de semana vou ver sobre o gatsby e o jekyll, ferramentas que não fiz nada ainda, pretendo fazer testes e seguir tutoriais para me iterar...

Mas a dúvida agora é referente ao next e gerar arquivos estáticos (para fazer deploy em um storage ou na netlify, estático mesmo), estou refazendo meu site para brincar.

Dúvida #1:
Configurei e gerei todas as páginas estáticas, ok, porém na navegação o client-side consome as apis de serviços do getInitialProps, ok, comportamento esperado, porém eu queria uma maneira de gerar tudo estático, inclusive estes dados que foram consultados na api serem consultados apenas em tempo de build mesmo, ao gerar o estático e ao navegar no client-side não quero que o react fique buscando os dados novamente na api, mas sim que o build já carregue os dados... Tem como? O que sugerem?

Dúvida 2:
Quando lidam com sites estáticos assim, sem consumo da api, o fluxo de deploy é este mesmo? A cada novo artigo ou novo dado gerar um novo build? Pretendo seguir assim ao menos neste projeto rápido que estou fazendo... 

Acredito que ao ver o jekyll e o gatsby talvez estas duas dúvidas ficaram mais claras, mas nestas dúvidas pensando no uso apenas do next.

Muito obrigado, abraços!

**Comment by matbrgz** [comment]
Infelizmente não posso te ajudar com Next, mas provavelmente existe uma forma, talvez medida alternativa (gambiarra). Mas em Jekyll é tudo estático, as páginas não utilizam um framework JS frontend (aka React), ou seja, as rotas são armazenadas em referência direta a página HTML.

Bem vindo ao mundo dos SSG's.

**Comment by alanhr** [comment]
Oi @wilsonneto-dev , não é possível você usar o `getInitialProps` com a  api do *next* servindo arquivos estáticos.

Resposta da duvida 1 :
Você pode consumir de uma api external ou criar uma **store** com esses dados

Resposta da duvida 2:
Normalmente sim, porque cada **artigo** é um novo `.md` que você precisa armazena no seu repositório.

**Comment by wilsonneto-swe** [comment]
Show, valeu @MatheusRV! É, comecei o curso de Jekyll e to gostando, ao menos para meu blog acho q vou fazer uma versão com jekyll para praticar e ver como é.

@alanhr valeu man, é, eu tentei até aqui mas essa bisnaga ta fazendo a consulta mesmo gerando estático, o getInitialProps ta sendo executado no client-side, o que é o certo, mas queria deixar td já no client/memória pra assim o site ser instântaneo quase... Até fiz um arquivão json que usei de base de dados, mas só com 30 artigos o .json já ficou gigante e a renderização no cliente ficou bem lenta, ou vai ser jekyll ou vou gerar stático no next mas remover o roteamento pelo react/client-side, deixar ele buscar as páginas no servidor mesmo, ou criar um json para cada página tambéme stático e ao no lugar de eu bater em uma api para que ela conecte no banco processando os dados e retornando o json, eu já gero esse json no build ou com uma function serverless.

Isso era mais uma POC para ver como otimizar além do SSR, também os dados.

Se alguém tiver mais sugestões rs

Abraços!
